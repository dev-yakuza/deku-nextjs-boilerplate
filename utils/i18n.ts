import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import { i18n as i18nConfig } from '../next-i18next.config'

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: Object.assign(
    {},
    ...i18nConfig.locales.map((locale) => ({
      [locale]: {
        common: require(`../public/locales/${locale}/common.json`),
      },
    })),
  ),
})

export default i18n
