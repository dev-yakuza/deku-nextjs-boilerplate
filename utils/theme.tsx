import { createTheme } from '@mui/material/styles'

export const theme = createTheme({
  palette: {
    primary: {
      main: '#ff8e88',
    },
  },
  typography: {
    h1: {
      fontSize: '2rem',
    },
  },
})
