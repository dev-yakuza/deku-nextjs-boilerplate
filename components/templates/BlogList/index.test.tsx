import { render } from '@testing-library/react'

import { mockGrid } from 'utils/test'

import { BlogList } from '.'

describe('<BlogList />', () => {
  it('Rendered well with no data', async () => {
    const { container } = render(<BlogList posts={undefined} />)

    const grid = mockGrid.mock.calls[0][0]
    expect(grid.container).toBe(true)
    expect(grid.children.length).toBe(2)

    const blogItem = grid.children[0]
    expect(blogItem).toEqual([])

    const floatingActionLink = grid.children[1]
    expect(floatingActionLink.type.name).toBe('FloatingActionLink')
    expect(floatingActionLink.props.link).toBe('/posts/add')
    expect(floatingActionLink.props.children.type.type.render.displayName).toBe(
      'AddIcon',
    )

    expect(container).toMatchSnapshot()
  })

  it('Rendered well with data', async () => {
    const { container } = render(
      <BlogList
        posts={[
          { id: 1, title: 'blog title 1', contents: 'blog contents 1' },
          { id: 2, title: 'blog title 2', contents: 'blog contents 2' },
          { id: 3, title: 'blog title 3', contents: 'blog contents 3' },
          { id: 4, title: 'blog title 4', contents: 'blog contents 4' },
          { id: 5, title: 'blog title 5', contents: 'blog contents 5' },
          { id: 6, title: 'blog title 6', contents: 'blog contents 6' },
          { id: 7, title: 'blog title 7', contents: 'blog contents 7' },
          { id: 8, title: 'blog title 8', contents: 'blog contents 8' },
        ]}
      />,
    )

    const grid = mockGrid.mock.calls[0][0]
    const blogItem = grid.children[0]
    expect(blogItem.length).toBe(8)
    expect(blogItem[0].props).toEqual({
      id: 1,
      title: 'blog title 1',
      contents: 'blog contents 1',
    })
    expect(blogItem[1].props).toEqual({
      id: 2,
      title: 'blog title 2',
      contents: 'blog contents 2',
    })
    expect(blogItem[2].props).toEqual({
      id: 3,
      title: 'blog title 3',
      contents: 'blog contents 3',
    })
    expect(blogItem[3].props).toEqual({
      id: 4,
      title: 'blog title 4',
      contents: 'blog contents 4',
    })
    expect(blogItem[4].props).toEqual({
      id: 5,
      title: 'blog title 5',
      contents: 'blog contents 5',
    })
    expect(blogItem[5].props).toEqual({
      id: 6,
      title: 'blog title 6',
      contents: 'blog contents 6',
    })
    expect(blogItem[6].props).toEqual({
      id: 7,
      title: 'blog title 7',
      contents: 'blog contents 7',
    })
    expect(blogItem[7].props).toEqual({
      id: 8,
      title: 'blog title 8',
      contents: 'blog contents 8',
    })

    const floatingActionLink = grid.children[1]
    expect(floatingActionLink.type.name).toBe('FloatingActionLink')

    expect(container).toMatchSnapshot()
  })
})
