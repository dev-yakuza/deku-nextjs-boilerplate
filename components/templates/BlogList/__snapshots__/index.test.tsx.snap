// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogList /> Rendered well with data 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) {
  @media (orientation: landscape) {
    .emotion-0 {
      min-height: 48px;
    }
  }
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 64px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  background-color: #fff;
  padding: 20px;
  margin: 20px;
  border-radius: 10px;
  box-shadow: 10px 10px 30px #d9d9d9,-10px -10px 30px #fff;
}

@media (min-width:600px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:900px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:1536px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-height: 36px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  padding: 0;
  min-width: 0;
  width: 56px;
  height: 56px;
  z-index: 1050;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
  color: rgba(0, 0, 0, 0.87);
  background-color: #e0e0e0;
  color: #fff;
  background-color: #1976d2;
  position: fixed;
  right: 40px;
  bottom: 40px;
}

.emotion-26::-moz-focus-inner {
  border-style: none;
}

.emotion-26.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-26 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-26:active {
  box-shadow: 0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12);
}

.emotion-26:hover {
  background-color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (hover: none) {
  .emotion-26:hover {
    background-color: #e0e0e0;
  }
}

.emotion-26.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-26.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-26:hover {
  background-color: #1565c0;
}

@media (hover: none) {
  .emotion-26:hover {
    background-color: #1976d2;
  }
}

.emotion-27 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-28 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular emotion-0"
  />
  <div
    class="MuiGrid-root MuiGrid-container emotion-1"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 1
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 1
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 2
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 2
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 3
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 3
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 4
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 4
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/5"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 5
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 5
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/6"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 6
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 6
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/7"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 7
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 7
        </p>
      </a>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <a
        href="/posts/8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog title 8
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          blog contents 8
        </p>
      </a>
    </div>
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary emotion-26"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-27"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-28"
      />
    </button>
  </div>
</div>
`;

exports[`<BlogList /> Rendered well with no data 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) {
  @media (orientation: landscape) {
    .emotion-0 {
      min-height: 48px;
    }
  }
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 64px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-height: 36px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  padding: 0;
  min-width: 0;
  width: 56px;
  height: 56px;
  z-index: 1050;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
  color: rgba(0, 0, 0, 0.87);
  background-color: #e0e0e0;
  color: #fff;
  background-color: #1976d2;
  position: fixed;
  right: 40px;
  bottom: 40px;
}

.emotion-2::-moz-focus-inner {
  border-style: none;
}

.emotion-2.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-2 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-2:active {
  box-shadow: 0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12);
}

.emotion-2:hover {
  background-color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: #e0e0e0;
  }
}

.emotion-2.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2:hover {
  background-color: #1565c0;
}

@media (hover: none) {
  .emotion-2:hover {
    background-color: #1976d2;
  }
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular emotion-0"
  />
  <div
    class="MuiGrid-root MuiGrid-container emotion-1"
  >
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary emotion-2"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</div>
`;
